### PR TITLE
Use current namespace and not hardcoded

### PR DESCRIPTION
**Pull request by rbo**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

/kind cleanup

**What this PR does / why we need it**:

Better deployment on single-namespace env. You don't have to change the deployment anymore.

Tested on openshift online ( openshift version 3.11 )

**Which issue(s) this PR fixes**:
https://github.com/tnozicka/openshift-acme/issues/115

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Single namespace and specific namespaces deployments now autodetect the controller namespace and don't require changing deployment fixtures for that.
```

### CODE REVIEW

File: deploy/single-namespace/deployment.yaml [hunk 1]
```diff
@@ -24,4 +24,9 @@ spec:
         args:
         - --exposer-image=quay.io/tnozicka/openshift-acme:exposer
         - --loglevel=4
-        - --namespace=acme-controller
+        - --namespace=$(CURRENT_NAMESPACE)
```
Comment (tnozicka) [inline deploy/single-namespace/deployment.yaml:27]: Thanks for sending the PR. Wanna get the other one as well?
https://github.com/tnozicka/openshift-acme/blob/master/deploy/specific-namespaces/deployment.yaml#L27

Comment (rbo) [reply deploy/single-namespace/deployment.yaml:27]: Yes, why not. Updated.

**Comment by tnozicka** [comment]
looks like the e2e flaked on openshift infra (image pull backoff)

to fix `verify` you need to run `make update`

File: deploy/specific-namespaces/deployment.yaml [hunk 2]
```diff
@@ -24,5 +24,10 @@ spec:
...
         - --exposer-image=quay.io/tnozicka/openshift-acme:exposer
         - --loglevel=4
-        - --namespace=acme-controller
+        - --namespace=$(CURRENT_NAMESPACE)
         - --namespace=test
+        env:
+        - name: CURRENT_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
```
Comment (tnozicka) [inline deploy/specific-namespaces/deployment.yaml:33]: nit: missing `\n`

Comment (rbo) [reply deploy/specific-namespaces/deployment.yaml]: oops, thanks! Fixed.

**Comment by tnozicka** [comment]
/retest
for the e2e flakes now that the registry fix merged https://github.com/tnozicka/openshift-acme/pull/118

passing verify requires you to run `make update` to update fixtures

**Comment by tnozicka** [comment]
/retest
looks like OCP 4.4 CI is having a bad day

**Comment by tnozicka** [comment]
@rbo e2es are green, can you please run `make update` to update generated fixtures and we can merge. thanks

**Comment by rbo** [comment]
Doesn't work on OSX:
```
 make update
Running `gofmt -s -l -w` on 34 file(s).
mv ./deploy/.diffs/* /var/folders/zg/djswqxd52v39541s7q8zntz80000gn/T/tmp.MGYh3JpH || true
hack/diff-deploy-files.sh ./deploy/.diffs
readlink: illegal option -- f
usage: readlink [-n] [file ...]
make: *** [update-deploy-files] Error 1
```

In a fedora container, it works :-)
```
$ docker run -ti -v $(pwd):/work fedora bash
# dnf install -y go make diffutils
# make update
Running `gofmt -s -l -w` on 34 file(s).
mv ./deploy/.diffs/* /tmp/tmp.vfGiTRTuNV || true
mv: cannot stat './deploy/.diffs/*': No such file or directory
hack/diff-deploy-files.sh ./deploy/.diffs
```

Updated and pushed.

**Comment by tnozicka** [comment]
Yeah, mac is always missing some options, this is kind of evergreen :( 

I think in the end the output also sometimes differ so containerized is the best option. Although we accept best effort PR for Mac compatibility.

Squash the first 3 commits and it is good to go.

/approve

**Comment by rbo** [comment]
Squashed...

**Comment by tnozicka** [comment]
@rbo thanks
/lgtm

**Comment by tnozicka** [comment]
/retest
openshift CI run into some issues today

**Comment by rbo** [comment]
looks like a stressful PR - sorry.

**Comment by tnozicka** [comment]
yeah, these flakes are coming from the actual OCP CI infra (openshift-acme test don't even start running), it's having a bad week
/retest

**Pull request merged**